authors/apps/articles/tests_articles/test_comments.py

```diff
@@ -0,0 +1,172 @@
+        c_id = self.comment_res.data['id']
+        self.thread_url = reverse("articles:create_thread",
+                                  args=(a_id, c_id,))
+
+    def test_create_comment(self):
+        """Test user can create a comment"""
+        response = self.client.post(self.url, self.comment_data,
+                                    format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_get_comments(self):
+        """Test user can get a comment"""
+        self.client.post(self.url, self.comment_data,
+                         format='json')
+        response = self.client.get(self.url,
+                                   format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_article_exists(self):
+        """Test article exists"""
+        self.res = self.client.post(
+            articles_url, self.data, format='json')
+        response = self.client.post('/api/articles/9/comments',
+                                    self.comment_data,
+                                    format='json')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_get_comment_body(self):
+        """ Test model method to get comment body """
+        self.comment_body = self.comment_data["comments"]["comment_body"]
+        comment = self.comment_body.__str__()
+        self.assertEqual(comment, "This is a comment")
+
+    def test_article_not_found(self):
+        """Test article are not found"""
+        response = self.client.get("/api/articles/11/comments",
+                                   self.data,
+                                   format='json')
+        self.assertIn("Sorry this article doesn\'t exist",
+                      str(response.data))
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_get_thread(self):
+        """ Test model method to get thread body """
+        self.thread = self.thread_data["threads"]["thread_body"]
+        thread = self.thread.__str__()
+        self.assertEqual(thread, "This is a thread comment")
+
+    def test_create_thread(self):
+        """Test whether a thread comment is created"""
+        response = self.client.post(self.thread_url,
+                                    self.thread_data,
+                                    format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_get_threads(self):
+        """Test whether thread comments are returned"""
+        self.client.post(self.thread_url,
+                         self.thread_data,
+                         format='json')
+        response = self.client.get(self.thread_url,
+                                   self.thread_data,
+                                   format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_comment_not_found(self):
```

Comment:
Why would a user want to get a single comment?

Reply:
This is to test when a user wants to comment on a comment(thread) and the parent comment was deleted or doesn't exist.